netmaster/k8snetwork/networkpolicy.go

```diff
@@ -135,204 +257,489 @@ func (k8sNet *k8sContext) deleteEpg(networkname, epgName, policyName string) err
+	} else { //Pods  belongs fromPart of Spec
+		fromPartPolicy := k8sNet.getMatchFromSpecPartNetPolicy(podInfo)
+		npLog.Infof("Delete Pod belong FromSec part:%+v",
+			fromPartPolicy)
+		if len(fromPartPolicy) > 0 {
+			npLog.Infof("remove PodIps:%v fromSpec part of Policy",
+				rmIps)
+			for _, nw := range fromPartPolicy {
+				k8sNet.rmIpFromSpecPodSelector(nw,
+					labelList, podInfo.IP)
+				rList := k8sNet.
+					buildIngressRuleToPodSelector(nw, rmIps,
+						nw.PodSelector.PolicyName)
+				npLog.Infof("Ingress Rule :%+v", *rList)
+				npLog.Infof("Pods Info in To Spec :%+v", rmIps)
+				ipList := getIpMapToSlice(nw.PodSelector.podIps)
+				ruleList := k8sNet.
+					finalIngressNetworkPolicyRule(nw,
+						ipList, *rList, false)
+				npLog.Infof("Pod rules:%+v", ruleList)
+			}
+		}
 	}
-	if err = k8sNet.deletePolicy(policyName); err != nil {
-		npLog.Errorf("failed to delete policy %s, %s", policyName, err)
-		return
+}
+func getIpMapToSlice(m map[string]string) []string {
+	ips := []string{}
+	for ip := range m {
+		ips = append(ips, ip)
+	}
+	return ips
+}
+func (k8sNet *k8sContext) UpdateIpListFromSpecfromLabel(nw *k8sNetworkPolicy,
+	label []string, ip string) {
+	for _, ingress := range nw.Ingress {
+		for _, podSelector := range ingress.IngressPodSelector {
+			for _, l := range label {
+				if ipMap, ok :=
+					podSelector.labelPodMap[l]; ok {
+					ipMap[ip] = true
+				}
+			}
+			//Rebuild PodSelector PodIps
+			k8sNet.updatePodSelectorPodIps(podSelector)
+			npLog.Infof("Update PodIps into FromSpecPod:%+v",
+				podSelector)
+		}
 	}
+	return
 }
 
-func (k8sNet *k8sContext) processK8sNetworkPolicy(opCode watch.EventType, np *v1.NetworkPolicy) {
-	if np.Namespace == "kube-system" { // not applicable for system namespace
-		return
+//Remove Give Pod Ips fromSpec Object of Network Policy
+func (k8sNet *k8sContext) rmIpFromSpecPodSelector(
+	nw *k8sNetworkPolicy, label []string, ip string) {
+	for _, ingress := range nw.Ingress {
+		for _, podSelector := range ingress.IngressPodSelector {
+			npLog.Infof("podSelector:%+v", podSelector)
+			//remove ips from PodSelector Object
+			delete(podSelector.podIps, ip)
+			for _, l := range label {
+				if ipMap, ok :=
+					podSelector.labelPodMap[l]; ok {
+					delete(ipMap, ip)
+					npLog.Infof("remove Pod Ips:%v FromSpec map:%+v",
+						ip, ipMap)
+				}
+			}
+			//Rebuild PodSelector PodIps
+			k8sNet.updatePodSelectorPodIps(podSelector)
+		}
 	}
+	return
+}
 
-	npLog.Infof("process [%s] network policy  %+v", opCode, np)
+//Add Pods Ips and readjuct Pod selector podIps list
+func (k8sNet *k8sContext) addPodIpsToSpecPodSelector(nw *k8sNetworkPolicy,
+	label []string, ip string) {
+	for _, l := range label {
+		if ipMap, ok := nw.PodSelector.labelPodMap[l]; ok {
+			ipMap[ip] = true
+		}
+	}
+	//Recalculate PodSelector PodIps list
+	for _, lMap := range nw.PodSelector.labelPodMap {
+		//At each Label Walk all its Ips
+		for ip := range lMap {
+			nw.PodSelector.podIps[ip] = ip
+		}
```

Comment:
another spot that might be implementing OR logic instead of AND logic for matchLabels